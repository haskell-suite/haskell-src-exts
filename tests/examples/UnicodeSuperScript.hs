{-# LANGUAGE UnicodeSyntax #-}

dumy ∷ Double → Double → Double
dumy a b =
    let a² = a ★ a
        b² = b ★ b
    in sqrt $ a² + b²
