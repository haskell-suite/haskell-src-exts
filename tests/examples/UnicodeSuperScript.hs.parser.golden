ParseOk (Module (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 1 8 1, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 1 1 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 1 4 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 8 1 8 1,SrcSpan "tests/examples/UnicodeSuperScript.hs" 8 1 8 1]}) Nothing [LanguagePragma (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 1 1 31, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 1 1 13,SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 28 1 31]}) [Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 1 14 1 27, srcInfoPoints = []}) "UnicodeSyntax"]] [] [TypeSig (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 1 3 35, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 6 3 8]}) [Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 1 3 5, srcInfoPoints = []}) "dumy"] (TyFun (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 9 3 35, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 16 3 18]}) (TyCon (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 9 3 15, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 9 3 15, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 9 3 15, srcInfoPoints = []}) "Double"))) (TyFun (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 19 3 35, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 26 3 28]}) (TyCon (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 19 3 25, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 19 3 25, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 19 3 25, srcInfoPoints = []}) "Double"))) (TyCon (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 29 3 35, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 29 3 35, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 3 29 3 35, srcInfoPoints = []}) "Double"))))),FunBind (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 1 7 22, srcInfoPoints = []}) [Match (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 1 7 22, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 1 4 5, srcInfoPoints = []}) "dumy") [PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 6 4 7, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 6 4 7, srcInfoPoints = []}) "a"),PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 8 4 9, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 8 4 9, srcInfoPoints = []}) "b")] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 10 7 22, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 4 10 4 11]}) (Let (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 5 7 22, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 5 5 8,SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 5 7 7]}) (BDecls (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 9 6 19, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 9 5 9,SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 9 6 9,SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 5 7 0]}) [PatBind (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 9 5 19, srcInfoPoints = []}) (PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 9 5 11, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 9 5 11, srcInfoPoints = []}) "a\178")) (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 12 5 19, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 12 5 13]}) (InfixApp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 14 5 19, srcInfoPoints = []}) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 14 5 15, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 14 5 15, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 14 5 15, srcInfoPoints = []}) "a"))) (QVarOp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 16 5 17, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 16 5 17, srcInfoPoints = []}) (Symbol (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 16 5 17, srcInfoPoints = []}) "*"))) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 18 5 19, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 18 5 19, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 5 18 5 19, srcInfoPoints = []}) "a"))))) Nothing,PatBind (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 9 6 19, srcInfoPoints = []}) (PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 9 6 11, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 9 6 11, srcInfoPoints = []}) "b\178")) (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 12 6 19, srcInfoPoints = [SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 12 6 13]}) (InfixApp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 14 6 19, srcInfoPoints = []}) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 14 6 15, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 14 6 15, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 14 6 15, srcInfoPoints = []}) "b"))) (QVarOp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 16 6 17, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 16 6 17, srcInfoPoints = []}) (Symbol (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 16 6 17, srcInfoPoints = []}) "*"))) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 18 6 19, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 18 6 19, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 6 18 6 19, srcInfoPoints = []}) "b"))))) Nothing]) (InfixApp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 8 7 22, srcInfoPoints = []}) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 8 7 12, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 8 7 12, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 8 7 12, srcInfoPoints = []}) "sqrt"))) (QVarOp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 13 7 14, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 13 7 14, srcInfoPoints = []}) (Symbol (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 13 7 14, srcInfoPoints = []}) "$"))) (InfixApp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 15 7 22, srcInfoPoints = []}) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 15 7 17, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 15 7 17, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 15 7 17, srcInfoPoints = []}) "a\178"))) (QVarOp (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 18 7 19, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 18 7 19, srcInfoPoints = []}) (Symbol (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 18 7 19, srcInfoPoints = []}) "+"))) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 20 7 22, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 20 7 22, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeSuperScript.hs" 7 20 7 22, srcInfoPoints = []}) "b\178"))))))) Nothing]],[])
